core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -270,22 +271,39 @@ func (r *EvmRegistry) refreshActiveUpkeeps() error {
 	}
 	r.active.Reset(ids...)
 
-	// refresh trigger config log upkeeps
-	// Note: We only update the trigger config for current active upkeeps and do not remove configs
-	// for any upkeeps that might have been removed. We depend upon state events to remove the configs
+	return r.refreshLogTriggerUpkeeps(ids)
+}
+
+// refreshLogTriggerUpkeeps refreshes the active upkeep ids for log trigger upkeeps
+//
+// TODO: manage reorgs, one idea is to pull the last ConfigSet event for each upkeep,
+// and to call updateTriggerConfig with latest config and block number.
+// Calling the log provider with newer block will update the filter accordingly.
+func (r *EvmRegistry) refreshLogTriggerUpkeeps(ids []*big.Int) error {
```

Comment:
I am not fully following the behaviour here. Want to confirm

- It removes filters for upkeeps that do not exist anymore
- For existing upkeeps it ignores them?  (Not sure how we can manage reorgs then?)
- For new upkeeps that somehow didn't have a filter, adds them with creation block 0 (should we use current latest block here?)

Reply:
`one idea is to pull the last ConfigSet event for each upkeep`

I realised we would also need to look for unpaused event